### PR TITLE
CI: add OSX building support for TravisCI builds

### PR DESCRIPTION
**Pull request by mrakitin**
## Description

Test OSX builds on TravisCI.

## Motivation and Context

Originally it was proposed via https://github.com/danielballan/bluesky/pull/18 - a PR to a PR branch, decided in a Slack conversation with @danielballan and @tacaswell it's worth to first test on the current master, and then suck it into #1125.

## How Has This Been Tested?

```bash
$ brew install travis. # or `brew upgrade travis`
$ travis lint
```

### CODE REVIEW

**Comment by mrakitin** [comment]
I guess
```
/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16
```
can be converted to
```
sudo Xvfb :99 -ac -screen 0 1280x1024x16 &
```
(sudo is set to false though in the beginning of the file.)

I saw in [#1125's branch](https://github.com/danielballan/bluesky/blob/8d982c379462d46c66453b0ade2a351069ffe771/.travis.yml#L5-L6) the following is in use: 
```yaml
services:
- xvfb
```

What do people think?

**Comment by mrakitin** [comment]
OSX Py35 build fails. I guess https://github.com/danielballan/bluesky/blob/8d982c379462d46c66453b0ade2a351069ffe771/.travis.yml#L14 can help (but can cause merging conflicts in #1125).

**Comment by tacaswell** [comment]
```
services:
- xvfb
```

is the way Travis wants you to do this now.

**Comment by tacaswell** [comment]
The 3.7 failures are fixed by un-pinning pyzmq.

**Comment by mrakitin** [comment]
> ```
> services:
> - xvfb
> ```
> 
> is the way Travis wants you to do this now.

@tacaswell, it does not work for every platform this way, only Ubuntu 16: https://docs.travis-ci.com/user/gui-and-headless-browsers/#using-services.

For OSX I followed this discussion: https://github.com/travis-ci/travis-ci/issues/7313#issuecomment-279837067.

**Comment by mrakitin** [comment]
The tests have passed! Thanks @tacaswell for the hints and your PR's `.travis.yml`. I think we can keep separate solutions for `Xvfb`'s for Linux and OSX for now. Or I can put `sudo Xvfb :99 -ac -screen 0 1280x1024x16 &` for both of them and see how it goes. Any preferences?

**Comment by danielballan** [comment]
I trust your judgement @mrakitin.

**Comment by mrakitin** [comment]
Hmm, I removed the xvfb completely, and the tests are still passing! See https://travis-ci.org/mrakitin/bluesky/builds/539025481 and https://github.com/mrakitin/bluesky/compare/add-osx-builds...mrakitin:tst-no-sudo-xvfb. Why did we need it at all? I guess a lot changed with the `xenial` dist + `qtbase5-dev` apt package + `Qt5Agg` backend.

It would be nice to incorporate this PR into #1125.

**Comment by mrakitin** [comment]
OK, this is passing without `Xvfb`. I think we can go ahead and use this configuration. Any volunteers to review/merge?

File: .travis.yml [hunk 1]
```diff
@@ -6,33 +7,66 @@ env:
...
   fast_finish: true
   include:
-    - python: 3.6
+    - os: linux
+      python: 3.6
       env: PUBLISH_DOCS=1
-    - python: nightly
+    - os: linux
+      python: 3.7
+      sudo: true
```
Comment (klauer) [inline .travis.yml]: Still necessary?

File: .travis.yml [hunk 2]
```diff
@@ -6,33 +7,66 @@ env:
...
   allow_failures:
     - python: nightly
 
 before_install:
-  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16"
-  - "export DISPLAY=:99.0"
+  - |
+    if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+        pushd ${HOME}
+        wget https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh && \
```
Comment (klauer) [inline .travis.yml]: I don't think it's necessary to chain the lines with ` && \` here. You might also consider removing some of the debug lines below.

File: .travis.yml (hunk 1 above)
Comment (mrakitin) [reply .travis.yml]: Good catch. I guess no.

Comment (mrakitin) [reply .travis.yml]: Addressed.

File: .travis.yml (hunk 2 above)
Comment (mrakitin) [reply .travis.yml]: Done. I left some `conda info/list` commands to have some information in which env we are running the tests.

**Comment by mrakitin** [comment]
All @klauer's suggestions were applied, and the tests pass. Feel free to merge if there are no other concerns.

**Comment by mrakitin** [comment]
Thanks, @klauer!

**Comment by tacaswell** [comment]
Ahh, this is probably why I have to rebase #1125 again .

**Comment by mrakitin** [comment]
Sorry @tacaswell

**Comment by danielballan** [comment]
Thanks again @mrakitin! This is great to have!

**Pull request merged**